ibis/backends/duckdb/tests/test_register.py

```diff
@@ -381,12 +354,45 @@ def test_memtable_with_nullable_pyarrow_not_string(con):
     assert len(res) == len(data)
 
 
-def test_set_temp_dir(tmp_path):
-    path = tmp_path / "foo" / "bar"
-    con = ibis.duckdb.connect(temp_directory=path, max_memory="1MB")
-    con.raw_sql("CREATE TABLE big AS SELECT range(100_000) as i")
-    # assert duckdb actually wrote something to this directory
-    assert list(path.iterdir())
+def _query_temp_dir_setting(con) -> Path:
+    query = "SELECT current_setting('temp_directory')"
+    return Path(con.raw_sql(query).fetchone()[0])
+
+
+def _trigger_oom(con):
+    con.raw_sql("SET memory_limit = '1MB'")
+    con.raw_sql("CREATE TABLE big AS SELECT range(1_000_000) as i")
+
+
+def test_temp_dir_unset_inmemory():
+    con = ibis.duckdb.connect()
+    p = _query_temp_dir_setting(con)
+    p.unlink(missing_ok=True)
+    _trigger_oom(con)
+    assert list(p.iterdir())
```

Comment:
These tests are incredibly flaky, since the behavior of setting the memory limit depends **greatly** on what else is running on the machine.

The best we can do here is to verify that `temp_directory` was set, but that's kind of pointless.

We should 100% avoid adding a bunch of tests that are verifying upstream default behavior.

Reply:
I'm going to remove most of these tests. I'll keep that ones that explicitly set the value of `temp_directory` to make sure that actually is being set on the pass-through. Everything else will be removed.